tests/test_minimal/test_utils/test_expand_paths.py

```diff
@@ -0,0 +1,75 @@
+import os
+from pathlib import Path
+
+from neuroconv.utils import LocalPathExpander
+
+
+#  helper functions to test for equivalence between set-like lists of dicts.
+def freeze(obj):
+    if isinstance(obj, dict):
+        return frozenset((k, freeze(v)) for k, v in obj.items())
+    elif isinstance(obj, list):
+        return frozenset(freeze(x) for x in obj)
+    return obj
+
+
+def are_equivalent_lists(list1, list2):
+    set1 = set(freeze(x) for x in list1)
+    set2 = set(freeze(x) for x in list2)
+    return set1 == set2
+
+
+def test_expand_paths(tmpdir):
+    expander = LocalPathExpander()
+
+    # set up directory for parsing
+    base = Path(tmpdir)
+    for subject_id in ("001", "002"):
+        Path.mkdir(base / f"sub-{subject_id}")
+        for session_id in ("101", "102"):
+            Path.mkdir(base / f"sub-{subject_id}" / f"session_{session_id}")
+            (base / f"sub-{subject_id}" / f"session_{session_id}" / "abc").touch()
+            (base / f"sub-{subject_id}" / f"session_{session_id}" / "xyz").touch()
+
+    # run path parsing
+    out = expander.expand_paths(
+        dict(
+            aa=dict(folder=base, file_path=os.path.join("sub-{subject_id:3}", "session_{session_id:3}", "abc")),
+            bb=dict(folder=base, file_path=os.path.join("sub-{subject_id:3}", "session_{session_id:3}", "xyz")),
```

Comment:
Note that if https://github.com/catalystneuro/neuroconv/pull/377/files#r1165963109 is accepted, this can simply become

```suggestion
            aa=dict(folder=base, file_path="sub-{subject_id:3}/session_{session_id:3}/abc"),
            bb=dict(folder=base, file_path="sub-{subject_id:3}/session_{session_id:3}/xyz"),
```

which is a more readable explicit test case